aya/src/programs/mod.rs

```diff
@@ -291,18 +316,22 @@ impl Program {
             Program::BtfTracePoint(p) => &mut p.data,
             Program::FEntry(p) => &mut p.data,
             Program::FExit(p) => &mut p.data,
+            Program::Extension(p) => &mut p.data,
         }
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub(crate) struct ProgramData {
+    pub(crate) name: Option<String>,
```

Comment:
I was using this so that `name` only gets set on `ProgramData` if we're going to use it for loading.
That test happens via the feature detection code when you construct a `BpfLoader`.
This is consistent with other optional load-time features - e.g `btf_fd`.

The alternative was for `load_program` to always load a test program first to figure out if loading a program name is supported or not... or to find a way to pass the detected features

Reply:
Ah, yeah ok that makes sense. I'm ok with landing this as is, we can refactor later